src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResendNewPinCallbackHandler.java

```diff
@@ -0,0 +1,138 @@
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+        this.claimIssuedNotificationService = claimIssuedNotificationService;
+        this.notificationsProperties = notificationsProperties;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_SUBMIT, this::resendNewPin
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse resendNewPin(CallbackParams callbackParams) {
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+        String authorisation = callbackParams.getParams().get(CallbackParams.Params.BEARER_TOKEN).toString();
+
+        if (getDefendantLinkStatus(claim).isLinked()) {
+            logger.info("Claim {} has already been linked to defendant - cannot send notification",
+                claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim has already been linked to defendant - cannot send notification"))
+                .build();
+        }
+
+        if (!claim.getClaimData().getDefendant().getEmail().isPresent()) {
+            logger.info("Claim {} doesn't have defendant email address - cannot send notification",
+                claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim doesn't have defendant email address - cannot send notification"))
+                .build();
+        }
+
+        GeneratePinResponse pinResponse = getPinResponse(claim.getClaimData(), authorisation)
+            .orElseThrow(() -> new IllegalArgumentException("Pin generation failed"));
+
+        claim.getClaimData().getDefendant().getEmail().ifPresent(defendantEmail ->
+            claimIssuedNotificationService.sendMail(
+                claim,
+                defendantEmail,
+                pinResponse.getPin(),
+                notificationsProperties.getTemplates().getEmail().getDefendantClaimIssued(),
+                "defendant-issue-notification-" + claim.getReferenceNumber(),
+                claim.getClaimData().getDefendant().getName()
+            ));
+
+        Claim updatedClaim = claim.toBuilder()
+            .letterHolderId(pinResponse.getUserId())
+            .build();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetailsConverter.convertToMap(caseMapper.to(updatedClaim)))
+            .build();
+    }
+
+    private Optional<GeneratePinResponse> getPinResponse(ClaimData claimData, String authorisation) {
+        return Optional.of(userService.generatePin(claimData.getDefendant().getName(), authorisation));
```

Comment:
Not sure why this is an optional. Does IDAM not throw an error if pin generation fails? Or does it 'succeed' but just return null?

Reply:
agree. I thought to handle if there is no pin generated for some reason but its a Feign Client.

Reply:
Based on swagger ui of IDAM, they never send null instead throw errors, so it should be fine to use directly.